packages/react-storage/src/components/StorageBrowser/controls/resolveComposable.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { useComposables } from '../composables/context';
+import { ComposableTypes } from '../composables/types';
+
+export function resolveComposable<
+  T extends React.ComponentType<any>,
+  K extends keyof ComposableTypes,
+>(
+  DefaultComponent: T,
+  name: K
+): (props: React.ComponentProps<T>) => React.JSX.Element {
+  const Component = (props: React.ComponentProps<T>) => {
+    const { components } = useComposables();
+    const ResolvedComponent = components?.[name] ?? DefaultComponent;
```

Comment:
Nit: the `component` and `composable` is a bit mixed-up IMO, should it be this?
```suggestion
    const { components: composables } = useComposables();
    const ResolvedComponent = composables?.[name] ?? DefaultComponent;
```

Or should useComposables return `composables` property instead?